Sources/NIO/AddressedEnvelope.swift

```diff
@@ -38,9 +38,11 @@ public struct AddressedEnvelope<DataType> {
     public struct Metadata: Hashable {
         /// Details of any congestion state.
         public var ecnState: NIOExplicitCongestionNotificationState
+        public let packetInfo: NIOPacketInfo?
         
-        public init(ecnState: NIOExplicitCongestionNotificationState) {
+        public init(ecnState: NIOExplicitCongestionNotificationState, packetInfo: NIOPacketInfo? = nil) {
```

Comment:
Unfortunately we can't add new elements to this initializer: it's not backwards compatible. Can you add a new initializer that takes two arguments instead of one? No need to default the second argument.